src/tokenize.rs

```diff
@@ -151,6 +151,10 @@ pub fn tokenize(input: &mut String) -> Result<RunnableBox, Box<dyn Error>> {
                     current_token.push(c);
                 } else if !escaped {
                     in_double_quotes = !in_double_quotes;
+                    if !in_double_quotes && !current_token.is_empty() {
+                        tokens.push(format!("LLM:{}", current_token));
+                        current_token.clear();
+                    }
```

Comment:
This doesn't discriminate between:

```bash
echo "hello"
```

and 

```bash
"hello"
```

Reply:
The current logic is if the user enters text between quotes on its own, it should be interpreted as a prompt for the LLM (i.e. `"hello"`).

This would result in just one element in the `tokens` vector.

However, other commands like `pwd` also result in only one element in the `tokens` vector. Therefore we need to distinguish between those two scenarios.

The difference is, of course, the fact that the LLM prompts are quoted.

So I think we should have some `DoubleQuote` struct that encapsulates the text between double quotes (similarly `SingleQuote`). This would then allow us to check if the `tokens` vector is of size 1 and contains a double-quoted string or not (right now the latter distinction is not possible to detect).

Having a `DoubleQuote` struct has the advantage of implementing #31 #30 #32 and other expansion-based operations that we could support in the future within double quotes.

Reply:
Doing this might be a bit involved, but hopefully not too bad.

Reply:
All fixed now 🚀